grpc_unary_attestation/src/server.rs

```diff
@@ -59,19 +134,75 @@ where
 impl<F, S, L> UnarySession for AttestationServer<F, L>
 where
     F: 'static + Send + Sync + Clone + FnOnce(Vec<u8>) -> S,
-    S: std::future::Future<Output = anyhow::Result<Vec<u8>>> + Send + Sync + 'static,
+    S: std::future::Future<Output = anyhow::Result<Vec<u8>>> + Send + Sync,
     L: Send + Sync + Clone + LogError + 'static,
 {
     async fn message(
         &self,
         request: tonic::Request<UnaryRequest>,
-    ) -> Result<tonic::Response<UnaryResponse>, tonic::Status> {
+    ) -> anyhow::Result<tonic::Response<UnaryResponse>, tonic::Status> {
         let error_logger = self.error_logger.clone();
+        let request_inner = request.into_inner();
+
+        let mut session_state = {
+            self.sessions_tracker
+                .lock()
+                .expect("Couldn't lock session_state mutex")
+                .pop_session_state(request_inner.session_id)
+                .map_err(|error| {
+                    error_logger.log_error(&error);
+                    tonic::Status::internal("")
```

Comment:
Are these error logged by tonic, or are they returned to the client? Would it help to just return the appropriate error and letting it be logged by the caller code, instead of returning empty errors here? Or would it leak information somehow?

Reply:
I think we never return any information to the client

Reply:
The `error_logger` logs server side, tonic just sends the empty status (mirrors the streaming client).

Without knowing the reasoning behind the initial implementation, I think not sending a specific error message is probably fine (and keeps us on the safe side security wise). In any case, after an error the client would attempt to restart the handshake.

Reply:
It is, they map over over to gRPC status codes: https://grpc.github.io/grpc/core/md_doc_statuscodes.html